### ISSUE TITLE
False positive for oas3-unused-component rule

### ISSUE THREAD

**Issue by crodriguezleon** [daveshanley/vacuum#442]
# Summary

Vacuum detects components as unused when they are only referenced from a component that inherits from another component using the `allOf` keyword.

# Example

```
components:
  schemas:
    Dog:
      type: object
      allOf:
        - $ref: '#/components/schemas/Pet'
      required:
        - breed
      properties:
        breed:
          $ref: '#/components/schemas/Breed'
    Pet:
      type: object
      properties:
       id: 
        type: String
        description: Unique identifier of the Pet
      age:
        type: integer
        format: int64
   Breed:
      type: object
      properties:
        name:
         type: String
       category:
        type: String
```

# Expected result
`Breed` should not be detected as unused by `oas3-unused-component` rule because it's being referenced from `Dog`.

# Actual result
Vacuum detects `Breed` as an unused component. If we remove the `allOf` section from the `Dog` definition, then `Breed` is not detected anymore as unused.

**Comment by daveshanley** [comment]
This has been resolved in `v0.8.7`

A test has been added to validate: 

https://github.com/daveshanley/vacuum/blob/main/functions/openapi/unused_component_test.go#L303

**Comment by crodriguezleon** [comment]
Hello @daveshanley,

Just tested it and the issue keeps happening for me with `v0.8.7`.

I've attached a zip file with the specification I've used to test it and the report I get. 

[Example.zip](https://github.com/daveshanley/vacuum/files/14162296/Example.zip)

Thanks in advance!

**Comment by daveshanley** [comment]
OK this is strange. I will re-investigate for you.

**Comment by daveshanley** [comment]
This has been resolved in `v0.9` 

The Example.zip now passes.
